glean-core/android/src/test/java/mozilla/telemetry/glean/pings/CustomPingTest.kt

```diff
@@ -0,0 +1,109 @@
+package mozilla.telemetry.glean.scheduler
+
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.telemetry.glean.Glean
+import mozilla.telemetry.glean.getContextWithMockedInfo
+import mozilla.telemetry.glean.getMockWebServer
+import mozilla.telemetry.glean.private.NoReasonCodes
+import mozilla.telemetry.glean.private.PingType
+import mozilla.telemetry.glean.resetGlean
+import mozilla.telemetry.glean.testing.GleanTestRule
+import mozilla.telemetry.glean.triggerWorkManager
+import mozilla.telemetry.glean.utils.tryGetLong
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.json.JSONObject
+import java.util.concurrent.TimeUnit
+
+/**
+ * Testing behavior of custom pings.
+ *
+ * We already rely on the Rust side to test custom pings,
+ * but this enables us to test the upload mechanism specifically.
+ *
+ * Even if this seemingly duplicates some of the testing, this should be kept around.
+ */
+@RunWith(AndroidJUnit4::class)
+class CustomPingTest {
+    @get:Rule
+    val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
```

Comment:
Can you use the [`GleanTestLocalServer`](https://searchfox.org/glean/source/glean-core/android/src/main/java/mozilla/telemetry/glean/testing/GleanTestLocalServer.kt) rule? here's [an example](https://searchfox.org/glean/source/samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/BaselinePingTest.kt#30).

Reply:
TIL.

Reply:
wait, this is for instrumented tests. This is not one of these, this is a plain "unit" test inside Glean itself.

Reply:
That won't work. I still need a glean around and I'd rather use the `GleanTestRule` for that. but that one also initializes work manager, and so does the TestLocalServer rule: https://github.com/mozilla/glean/blob/m16/concurrent-scheduling/glean-core/android/src/main/java/mozilla/telemetry/glean/testing/GleanTestLocalServer.kt#L48-L54